central/productusage/datastore/securedunits/datastore.go

```diff
@@ -0,0 +1,36 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/stackrox/rox/central/productusage/source"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// DataStore is the datastore for usage.
+//
+//go:generate mockgen-wrapper
+type DataStore interface {
+	// Persistent storage access:
+
+	// Get returns the channel, from which the metrics could be read.
+	Get(ctx context.Context, from *types.Timestamp, to *types.Timestamp) (<-chan *storage.SecuredUnits, error)
+	// Insert puts metrics to the persistent storage.
+	Insert(ctx context.Context, metrics *storage.SecuredUnits) error
+
+	// In-memory storage access:
+
+	// AggregateAndFlush returns the aggregated metrics from the
+	// in-memory storage and resets the storage.
+	AggregateAndFlush(ctx context.Context) (*storage.SecuredUnits, error)
+	// GetCurrentUsage returns the currently known usage.
+	GetCurrentUsage(ctx context.Context) (*storage.SecuredUnits, error)
+	// UpdateUsage updates the in-memory storage with the cluster metrics.
+	UpdateUsage(ctx context.Context, clusterID string, metrics source.SecuredUnitsSource) error
```

Comment:
Why `UpdateUsage()` and only it returns `source.SecuredUnitsSource` while other functions return `storage.SecuredUnits`?

Reply:
Usage driven inconsistency. `GetCurrentUsage` is used by the service, and `UpdateUsage` is called from the sensor clustermetrics pipeline, where the data is provided within internal `central.ClusterMetrics` storage object. I didn't want to depend on the latter here, nor on the product usage storage object in the pipeline code.
I'll make this interface consistent, in the hope it will be refactored to avoid using storage structures at all.